File: testdata/go/expected/pkg/outputtest/services.go
package outputtest

import (
	"context"

	"github.com/go-logr/logr"
	"github.com/google/uuid"
)

type MyServiceImpl struct {
	log        logr.Logger
	repository Repository
}

func NewMyServiceImpl(log logr.Logger, repository Repository) *MyServiceImpl {
	return &MyServiceImpl{
		log:        log,
		repository: repository,
	}
}

func (s *MyServiceImpl) EmptyVoid(ctx context.Context) error {
	return nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryType(ctx context.Context, value *MyType) (*MyType, error) {
	return &MyType{}, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryEnum(ctx context.Context, value MyEnum) (MyEnum, error) {
	return MyEnum(0), nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryAlias(ctx context.Context, value uuid.UUID) (uuid.UUID, error) {
	return uuid.UUID{}, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryString(ctx context.Context, value string) (string, error) {
	return "", nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryI64(ctx context.Context, value int64) (int64, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryI32(ctx context.Context, value int32) (int32, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryI16(ctx context.Context, value int16) (int16, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryI8(ctx context.Context, value int8) (int8, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryU64(ctx context.Context, value uint64) (uint64, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryU32(ctx context.Context, value uint32) (uint32, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryU16(ctx context.Context, value uint16) (uint16, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryU8(ctx context.Context, value uint8) (uint8, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryF64(ctx context.Context, value float64) (float64, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryF32(ctx context.Context, value float32) (float32, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) UnaryBytes(ctx context.Context, value []byte) ([]byte, error) {
	return []byte{}, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncType(ctx context.Context, value *MyType, optional *MyType) (*MyType, error) {
	return &MyType{}, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncEnum(ctx context.Context, value MyEnum, optional *MyEnum) (MyEnum, error) {
	return MyEnum(0), nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncAlias(ctx context.Context, value uuid.UUID, optional *uuid.UUID) (uuid.UUID, error) {
	return uuid.UUID{}, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncString(ctx context.Context, value string, optional *string) (string, error) {
	return "", nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncI64(ctx context.Context, value int64, optional *int64) (int64, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncI32(ctx context.Context, value int32, optional *int32) (int32, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncI16(ctx context.Context, value int16, optional *int16) (int16, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncI8(ctx context.Context, value int8, optional *int8) (int8, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncU64(ctx context.Context, value uint64, optional *uint64) (uint64, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncU32(ctx context.Context, value uint32, optional *uint32) (uint32, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncU16(ctx context.Context, value uint16, optional *uint16) (uint16, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncU8(ctx context.Context, value uint8, optional *uint8) (uint8, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncF64(ctx context.Context, value float64, optional *float64) (float64, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncF32(ctx context.Context, value float32, optional *float32) (float32, error) {
	return 0, nil // TODO: Provide implementation.
}

func (s *MyServiceImpl) FuncBytes(ctx context.Context, value []byte, optional []byte) ([]byte, error) {
	return []byte{}, nil // TODO: Provide implementation.
}

File: testdata/cs/expected/pkg/outputtest/services.cs
namespace Apex.Testing
{
	public class MyServiceImpl : MyService
	{
		private Repository repository;

		public MyServiceImpl (RepositoryImpl repository)
		{
			this.repository = repository;
		}

		public void EmptyVoid()
		{
			return; // TODO: Provide implementation.
		}

		public MyType UnaryType(MyType value)
		{
			return new MyType(); // TODO: Provide implementation.
		}

		public MyEnum UnaryEnum(MyEnum value)
		{
			return new MyEnum(); // TODO: Provide implementation.
		}

		public Uuid UnaryAlias(Uuid value)
		{
			return new Uuid(); // TODO: Provide implementation.
		}

		public string UnaryString(string value)
		{
			return new string(); // TODO: Provide implementation.
		}

		public long UnaryI64(long value)
		{
			return new long(); // TODO: Provide implementation.
		}

		public int UnaryI32(int value)
		{
			return new int(); // TODO: Provide implementation.
		}

		public short UnaryI16(short value)
		{
			return new short(); // TODO: Provide implementation.
		}

		public sbyte UnaryI8(sbyte value)
		{
			return new sbyte(); // TODO: Provide implementation.
		}

		public ulong UnaryU64(ulong value)
		{
			return new ulong(); // TODO: Provide implementation.
		}

		public uint UnaryU32(uint value)
		{
			return new uint(); // TODO: Provide implementation.
		}

		public ushort UnaryU16(ushort value)
		{
			return new ushort(); // TODO: Provide implementation.
		}

		public byte UnaryU8(byte value)
		{
			return new byte(); // TODO: Provide implementation.
		}

		public double UnaryF64(double value)
		{
			return new double(); // TODO: Provide implementation.
		}

		public float UnaryF32(float value)
		{
			return new float(); // TODO: Provide implementation.
		}

		public byte[] UnaryBytes(byte[] value)
		{
			return new byte[](); // TODO: Provide implementation.
		}

		public MyType FuncType(MyType value, MyType? optional)
		{
			return new MyType(); // TODO: Provide implementation.
		}

		public MyEnum FuncEnum(MyEnum value, MyEnum? optional)
		{
			return new MyEnum(); // TODO: Provide implementation.
		}

		public Uuid FuncAlias(Uuid value, Uuid? optional)
		{
			return new Uuid(); // TODO: Provide implementation.
		}

		public string FuncString(string value, string? optional)
		{
			return new string(); // TODO: Provide implementation.
		}

		public long FuncI64(long value, long? optional)
		{
			return new long(); // TODO: Provide implementation.
		}

		public int FuncI32(int value, int? optional)
		{
			return new int(); // TODO: Provide implementation.
		}

		public short FuncI16(short value, short? optional)
		{
			return new short(); // TODO: Provide implementation.
		}

		public sbyte FuncI8(sbyte value, sbyte? optional)
		{
			return new sbyte(); // TODO: Provide implementation.
		}

		public ulong FuncU64(ulong value, ulong? optional)
		{
			return new ulong(); // TODO: Provide implementation.
		}

		public uint FuncU32(uint value, uint? optional)
		{
			return new uint(); // TODO: Provide implementation.
		}

		public ushort FuncU16(ushort value, ushort? optional)
		{
			return new ushort(); // TODO: Provide implementation.
		}

		public byte FuncU8(byte value, byte? optional)
		{
			return new byte(); // TODO: Provide implementation.
		}

		public double FuncF64(double value, double? optional)
		{
			return new double(); // TODO: Provide implementation.
		}

		public float FuncF32(float value, float? optional)
		{
			return new float(); // TODO: Provide implementation.
		}

		public byte[] FuncBytes(byte[] value, byte[]? optional)
		{
			return new byte[](); // TODO: Provide implementation.
		}
	}

}
